packages/web-config-server/src/aggregator/QueryBuilder.js

```diff
@@ -67,8 +67,10 @@ export class QueryBuilder {
       // If a the api consumer defined a period to use in query parameters, use that
       this.query.period = this.replacementValues.period;
     } else if (!this.query.period) {
-      // If no period defined anywhere, use the default
-      this.query.period = getDefaultPeriod();
+      // If no period defined anywhere, use the period defined by the start and end dates or the default
+      this.query.period =
```

Comment:
That's an excellent point! From this code snippet it seems like startDate and endDate are supported in `DhisService`, `TupaiaService`, `WeatherService` (although it looks like `period` is NOT supported) and `IndicatorService` (as the `fetchOptions` are just passed through)

This code snippet seems to have caused the problem (the startDate and endDate are ignored if the period is set).
```
const addTemporalDimension = (query, { period, startDate, endDate }) =>
  period
    ? { ...query, dimension: (query.dimension || []).concat(`pe:${period}`) }
    : { ...query, startDate, endDate };
```

Are you suggesting that we consolidate and only send through `startDate` and `endDate`? If so, my only concern is that we theoretically lose the ability to fetch data for disjoint periods in one call, but I'm not sure if that's worth considering.

Reply:
Thanks for looking into that. 

Good point about the disjoint periods in one call, though I'm not sure we've ever used that, so I'd suggest we shouldn't maintain support for it just because one day we might (YAGNI, as Igor is fond of saying)

I would prefer to consolidate completely, but I'm not sure that's possible - do all DhisApi functions support start and end date? If they do, I'm keen. If not, it'd be nice to alter them to have that support, though it would be extending the scope too much to fit into this PR.

Your change is definitely an improvement and I'm happy to approve it if anything _needs_ period.

Reply:
I can't find anything that needs period..., but I also can't prove that nothing needs it? I guess if this will be regression tested anyway then it's ok?

Reply:
Just looking through and from a hazy memory, I think `getAnalytics` might use it (and that's why `period` is preferred in that util). 

Could you have a quick look into that, but assuming analytics breaks if `period` isn't provided, let's just leave your code as is here.

Reply:
I should have specified, that `addTemporalDimension` util is used in both event analytics and dataValue analytics...

Additionally, both api's look like they support it:
![image](https://user-images.githubusercontent.com/32168886/107595180-0b7be680-6c68-11eb-9afe-62bbc694a600.png)
![image](https://user-images.githubusercontent.com/32168886/107595435-ae346500-6c68-11eb-9bbc-423945ca18a4.png)

Saying that, I'm still more than happy to leave it as it is.

Reply:
Oh that's good! Cool, well...it would clean things up. Let's try to consolidate and regression test. Maybe worth a final check that @kael89 doesn't know any place that needs `period` downstream from aggregator?

Reply:
The only place I can think of where `period` is used in a different way than `startDate`/`endDate` is potentially the `sumPreviousPerPeriod` aggregation (requestedPeriod is used there)

Reply:
Sweet, and `requestedPeriod` could be converted to a `startDate`/`endDate` too if needed right?

I've just checked, and DHIS2 added support for `startDate`/`endDate` in v2.31, which I didn't cotton on to at the time. This is great, thanks for prompting us to discover that @Alastair-L 🎉

Reply:
Can I leave this for another PR (I'm strictly meant to be on Tamanu today)?

Reply:
Yep :+1: